tests/e2e/e2e_util_test.go

```diff
@@ -180,14 +187,101 @@ func (s *IntegrationTestSuite) submitProposal(c *chain.Chain, upgradeHeight int)
+	s.T().Log("successfully submitted upgrade proposal")
+	c.PropNumber = c.PropNumber + 1
+}
+
+func (s *IntegrationTestSuite) submitSuperfluidProposal(c *chain.Chain, asset string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("submitting superfluid proposal for asset %s on %s container: %s", asset, s.valResources[c.ChainMeta.Id][0].Container.Name[1:], s.valResources[c.ChainMeta.Id][0].Container.ID)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	s.Require().Eventually(
+		func() bool {
+			exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+				Context:      ctx,
+				AttachStdout: true,
+				AttachStderr: true,
+				Container:    s.valResources[c.ChainMeta.Id][0].Container.ID,
+				User:         "root",
+				Cmd: []string{
+					"osmosisd", "tx", "gov", "submit-proposal", "set-superfluid-assets-proposal", fmt.Sprintf("--superfluid-assets=%s", asset), fmt.Sprintf("--title=\"%s superfluid asset\"", asset), fmt.Sprintf("--description=\"%s superfluid asset\"", asset), "--from=val", "-b=block", "--yes", "--keyring-backend=test", "--log_format=json", fmt.Sprintf("--chain-id=%s", c.ChainMeta.Id),
+				},
+			})
+			s.Require().NoError(err)
+
+			err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+				Context:      ctx,
+				Detach:       false,
+				OutputStream: &outBuf,
+				ErrorStream:  &errBuf,
+			})
+			return strings.Contains(outBuf.String(), "code: 0")
+		},
+		time.Minute,
+		time.Second,
+		"tx returned a non-zero code; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	s.T().Log("successfully submitted superfluid proposal")
+	c.PropNumber = c.PropNumber + 1
+}
+
+func (s *IntegrationTestSuite) submitTextProposal(c *chain.Chain, text string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("submitting text proposal on %s container: %s", s.valResources[c.ChainMeta.Id][0].Container.Name[1:], s.valResources[c.ChainMeta.Id][0].Container.ID)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	s.Require().Eventually(
+		func() bool {
+			exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+				Context:      ctx,
+				AttachStdout: true,
+				AttachStderr: true,
+				Container:    s.valResources[c.ChainMeta.Id][0].Container.ID,
+				User:         "root",
+				Cmd: []string{
+					"osmosisd", "tx", "gov", "submit-proposal", "--type=text", fmt.Sprintf("--title=\"%s\"", text), "--description=\"test text proposal\"", "--from=val", "-b=block", "--yes", "--keyring-backend=test", "--log_format=json", fmt.Sprintf("--chain-id=%s", c.ChainMeta.Id),
+				},
+			})
+			s.Require().NoError(err)
+
+			err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+				Context:      ctx,
+				Detach:       false,
+				OutputStream: &outBuf,
+				ErrorStream:  &errBuf,
+			})
+			return strings.Contains(outBuf.String(), "code: 0")
```

Comment:
Lets refactor all of this into a helper function, this is too much boilerplate to be repeating so much

Reply:
On skim, it appears to me that line 242 through 271 an be refactored into a single helper function, and re-used in most of these tests.

Reply:
Can later make a separate helper method that lets for more control of various arguments, e.g. timeouts and container id

Reply:
If you prefer to keep the diff for this PR small, perhaps add the helper, and then just use it for new tests, and make an issue  and/or second PR for using it for everything else.

(I'm also happy with using it for everything in this PR as well though)

Reply:
You can even make the `Cmd` in the helper just be a string thats formatted, and the helper can handle splitting it on spaces.

Reply:
@p0mvn and I had discussed moving all of this boilerplate to a separate function this morning, but we first wanted to ensure we got something that just works prior to the upgrade (we were hitting a bunch of issues getting our tests to work in the first place). Now that this is in, we can go back and redo this (we can do it pre PR merge as well, I have no issue with that)